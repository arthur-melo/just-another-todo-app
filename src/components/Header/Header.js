import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faListAlt } from '@fortawesome/free-solid-svg-icons';

import './Header.css';

const headerPropTypes = {
  className: PropTypes.string,
};

const Header = ({ className }) => (
  <div className={className}>
    <div className="header__container">
      <nav className="navbar">
        <span className="header__span navbar-brand">
          <FontAwesomeIcon className="d-inline-block" icon={faListAlt} size="lg" alt="List emoji" />
        </span>
        <span className="header__span">Just Another Todo App</span>
      </nav>
    </div>
  </div>
);

Header.propTypes = headerPropTypes;

export default Header;
