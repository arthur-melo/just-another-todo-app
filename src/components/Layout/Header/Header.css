.header__text {
  margin-left: 0.3em;
  color: var(--bs-primary);
}
