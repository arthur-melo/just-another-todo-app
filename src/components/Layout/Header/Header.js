import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faListAlt } from '@fortawesome/free-solid-svg-icons';

import './Header.css';
import useTheme from '../../../hooks/useTheme';

const Header = () => {
  const [colors] = useTheme();

  return (
    <div className="header__container">
      <nav className="navbar">
        <span className="header__span navbar-brand">
          <FontAwesomeIcon
            className="d-inline-block"
            icon={faListAlt}
            size="lg"
            alt="List emoji"
            color={colors.primary}
          />
        </span>
        <span className="header__span">Just Another Todo App</span>
      </nav>
    </div>
  );
};

export default Header;
