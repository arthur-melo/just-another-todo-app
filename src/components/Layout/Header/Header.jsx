import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faListAlt } from '@fortawesome/free-solid-svg-icons';

import './Header.css';
import useTheme from '../../../hooks/useTheme';

const Header = () => {
  const [colors] = useTheme();

  return (
    <div className="d-flex justify-content-center align-items-center w-100 h-100 bg-light">
      <nav className="navbar">
        <span className="navbar-brand m-0">
          <FontAwesomeIcon
            className="d-inline-block"
            icon={faListAlt}
            size="lg"
            alt="List emoji"
            color={colors.primary}
          />
        </span>
        <span className="header__text align-top fs-5">
          Just Another Todo App
        </span>
      </nav>
    </div>
  );
};

export default Header;
