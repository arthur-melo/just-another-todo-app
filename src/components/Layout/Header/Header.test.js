import renderer from 'react-test-renderer';

import Header from './Header';

describe('Header', () => {
  it('should render the Header component', () => {
    const view = renderer.create(<Header />).toJSON();
    expect(view).toMatchSnapshot();
  });
});
