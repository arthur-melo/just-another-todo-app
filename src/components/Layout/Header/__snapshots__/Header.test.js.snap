// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the Header component 1`] = `
<div
  className="d-flex justify-content-center align-items-center w-100 h-100 bg-light"
>
  <nav
    className="navbar"
  >
    <span
      className="navbar-brand m-0"
    >
      <svg
        alt="List emoji"
        aria-hidden="true"
        className="svg-inline--fa fa-rectangle-list fa-lg d-inline-block"
        data-icon="rectangle-list"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 60.7 28.7 32 64 32H512c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V96zM128 288c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32zm32-128c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zM128 384c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32zm96-248c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    <span
      className="header__text align-top fs-5"
    >
      Just Another Todo App
    </span>
  </nav>
</div>
`;
